plugins/modules/system/xfconf.py

```diff
@@ -299,7 +301,7 @@ def main():
         module.exit_json(changed=True, ansible_facts=facts)
 
     except Exception as e:
-        module.fail_json(msg="Failed with exception: {0}".format(e))
+        module.fail_json(msg="Failed with exception: {0}".format(e), exception=to_text(traceback.format_exc()))
```

Comment:
I didn't find one single case in an existing module where `traceback.format_exc()` was converted to anything before being returned. Is there a reason why you prefer `to_text` here?
```suggestion
        module.fail_json(msg="Failed with exception: {0}".format(e), exception=traceback.format_exc())
```

Reply:
Because of your previous comment, I got myself into the ansible code a little bit. I wanted to know how ansible handled exceptions, and it turns out it does have a try/except block around task execution:

https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_executor.py#L194-L195

Nonetheless, I thought it over and it is probably better to have an explicit handler than an implicit one.

So, I used `to_text` because I copied from that line.

Reply:
Happy to remove it though

Reply:
That's controller code, not module code :) (see https://gist.github.com/sivel/9eb368f98974e15033c3ac2f989ac4e0#on-the-target) Also, that `try`/`except` is not for catching exceptions in modules, but in action plugins. The code which reads the output of crashed modules is here: https://github.com/ansible/ansible/blob/6894ae7d1d5feeae6690435bf8d428fa4eeb5463/lib/ansible/plugins/action/__init__.py#L1149